sdk/src/compressed_account.rs

```diff
@@ -636,3 +429,34 @@ pub fn pack_compressed_account(
 ) -> PackedCompressedAccountWithMerkleContext {
     pack_compressed_accounts(&[compressed_account], &[root_index], remaining_accounts)[0].clone()
 }
+
+pub fn pack_light_account_infos(
+    light_account_infos: &[LightAccountInfo],
+    remaining_accounts: &mut RemainingAccounts,
+) -> Vec<PackedLightAccountInfo> {
+    light_account_infos
+        .iter()
+        .map(|x| PackedLightAccountInfo {
+            compressed_account: x.compressed_account.clone(),
+            input_merkle_context: pack_merkle_context(x.input_merkle_context, remaining_accounts),
+            output_merkle_context: pack_merkle_context(x.output_merkle_context, remaining_accounts),
+        })
+        .collect::<Vec<_>>()
+}
+
+pub fn pack_light_account_info(
+    light_account_info: LightAccountInfo,
+    remaining_accounts: &mut RemainingAccounts,
+) -> PackedLightAccountInfo {
+    PackedLightAccountInfo {
+        compressed_account: light_account_info.compressed_account,
```

Comment:
we shouldn't send the complete compressed account as instruction data only the pure data `light_account_info.compressed_account.data.data`.

`PackedLightAccountInfo` should probably have a separate optional owner field. If owner is None it is owned by the program that is being invoked, if owner is some it is owned by a different program which is invoked via cpi during program runtime hence the account must be in the accounts of the instruction and we can just point to the position of the account in the accounts slice (pointing to the position of the account is easier said than done since at least in anchor we don't have access to the lowest level account_info array).

Reply:
Is this abstraction used anywhere in this pr?